<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Epidemiology for Bioinformaticians • epiforbioworkshop2022</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Epidemiology for Bioinformaticians">
<meta property="og:description" content="epiforbioworkshop2022">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">epiforbioworkshop2022</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.6</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/EpiForBioWorkshop.html">Epidemiology for Bioinformaticians</a>
    </li>
    <li>
      <a href="../articles/questionResponses.html">Answers to BioC2020 Questions</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://chloemirzayi.com/epiforbioworkshop2022">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Epidemiology for Bioinformaticians</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/cmirzayi/epiforbioworkshop2022/blob/HEAD/vignettes/EpiForBioWorkshop.Rmd" class="external-link"><code>vignettes/EpiForBioWorkshop.Rmd</code></a></small>
      <div class="hidden name"><code>EpiForBioWorkshop.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="instructors-names-and-contact-information">Instructor(s) name(s) and contact information<a class="anchor" aria-label="anchor" href="#instructors-names-and-contact-information"></a>
</h2>
<p>Chloe Mirzayi, <a href="mailto:cmirzayi@gmail.com" class="email">cmirzayi@gmail.com</a></p>
<p>Levi Waldron, <a href="mailto:levi.waldron@sph.cuny.edu" class="email">levi.waldron@sph.cuny.edu</a></p>
<p>CUNY School of Public Health, 55 W 125th St, New York, NY 10027
USA</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/waldronlab/curatedMetagenomicData" class="external-link">curatedMetagenomicData</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://www.dagitty.net" class="external-link">dagitty</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/mikelove/DESeq2" class="external-link">DESeq2</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://dx.plos.org/10.1371/journal.pone.0061217" class="external-link">phyloseq</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/malcolmbarrett/ggdag" class="external-link">ggdag</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/microbiome/mia" class="external-link">mia</a></span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="smoking-and-lung-cancer-the-most-famous-causal-inference-problem">Smoking and lung cancer: the most famous causal inference
problem<a class="anchor" aria-label="anchor" href="#smoking-and-lung-cancer-the-most-famous-causal-inference-problem"></a>
</h2>
<p>Though the idea that smoking causes lung cancer is a well-accepted
fact these days, it was an incredibly controversial theory in the 1950s
and 60s. The statistician RA Fisher (who developed ANOVA, the
F-distribution, and more) believed that smoking <strong><em>did
not</em></strong>cause lung cancer, arguing that correlation does not
necessarily mean causation.</p>
<p>Questions to consider during this workshop:</p>
<ol style="list-style-type: decimal">
<li>If you wanted to prove Fisher wrong and demonstrate smoking causes
lung cancer, how would you do so?</li>
<li>One of Fisher’s main arguments was that there was a genetic factor
that was correlated with both smoking behaviors and lung cancer. He
argued that this genetic factor was what was truly causing lung cancer
in smokers. How would you test this theory?</li>
</ol>
</div>
<div class="section level2">
<h2 id="goals-and-outline">Goals and Outline<a class="anchor" aria-label="anchor" href="#goals-and-outline"></a>
</h2>
<ul>
<li><p>Introduce fundamental concepts of causal inference</p></li>
<li><p>Discuss how to make directed acyclic graphs (DAGs) and how to use
them when analyzing data</p></li>
<li><p>Explore concepts of bias and confounding</p></li>
<li><p>Review common study designs and their implications for causal
inference</p></li>
<li><p>Briefly survey tools in R that can help with causal
inference</p></li>
</ul>
</div>
<div class="section level2">
<h2 id="introduction-just-what-is-a-cause-anyway">Introduction: Just what is a cause anyway?<a class="anchor" aria-label="anchor" href="#introduction-just-what-is-a-cause-anyway"></a>
</h2>
<p><strong>Question:</strong> In your own words, what is a cause? What
does it mean to cause something?</p>
<p>How do we know our potential cause brought about an effect or a
result? Perhaps it preceded the effect of interest. However, it could
still just be a coincidence. Perhaps there is a lurking variable that
truly causes both what we think is the cause and the observed
effect.</p>
<p>Despite this, humans (and animals) tend to have a relatively
intuitive understanding of causation when we observe it directly. When I
burn my finger while cooking, I know what caused it–my finger coming
into contact with a hot stove. I could also trace the causal pathway
backward and identify what caused my finger to come into contact with
the hot stove in the first place–such as my motivation to cook dinner or
me being distracted by the antics of my cat.</p>
<p><strong>Question:</strong> What are some causal questions we may be
interested in within the context of Bioinformatics (or your field of
work/study/research)?</p>
</div>
<div class="section level2">
<h2 id="the-counterfactual-definition-of-a-cause">The counterfactual definition of a cause<a class="anchor" aria-label="anchor" href="#the-counterfactual-definition-of-a-cause"></a>
</h2>
<p>However, the nature of much of the work I and many Bioinformaticians
do does not allow us to directly observe what caused an event of
interest. Instead we must rely on inferences we draw from our data to
establish an argument for a particular causal mechanism or pathway.</p>
<p>Rather than relying on vague dictionary definitions to inform the
science of causal inference, the most commonly accepted definition of a
cause in modern epidemiology is based on a counterfactual: <em>what
would have happened had the event of interest had the exposure not
occurred</em>.</p>
<p>But that’s not very intuitive or easy to parse so let’s consider an
example:</p>
<div class="section level3">
<h3 id="example-a-dog-and-an-ambulance">Example: A dog and an ambulance<a class="anchor" aria-label="anchor" href="#example-a-dog-and-an-ambulance"></a>
</h3>
<p>An ambulance drives by a house with its sirens on. A dog in the
house’s yard barks. We can ask the causal question in a straightforward
way: <strong>Did the ambulance driving by the house cause the dog to
bark?</strong></p>
<p>Alternatively, we can rephrase the question in terms of a
counterfactual: Would the dog have barked if the ambulance had not
driven by the house?</p>
</div>
<div class="section level3">
<h3 id="more-examples-of-causal-questions-rephrased-as-counterfactuals">More examples of causal questions rephrased as counterfactuals<a class="anchor" aria-label="anchor" href="#more-examples-of-causal-questions-rephrased-as-counterfactuals"></a>
</h3>
<p>“Does the vaccine increase the risk of blood clots?” -&gt; “Would
people have developed blood clots if they had not been given the
vaccine?”</p>
<p><strong>Question:</strong> Using the causal question from
Bioinformatics or your field, reframe it in terms of a
counterfactual?</p>
</div>
</div>
<div class="section level2">
<h2 id="dags">DAGs<a class="anchor" aria-label="anchor" href="#dags"></a>
</h2>
<p>We can visualize the relationship using a causal diagram also called
a directed acyclic graph (DAG).</p>
<p><img src="EpiForBioWorkshop_files/figure-html/unnamed-chunk-2-1.png" width="288"></p>
<p>This diagram depicts the cause or exposure (the ambulance) and the
effect or the outcome (the dog barking) with an arrow indicating the
direction of the causal relationship. In contrast to a diagram showing a
statistical relationship, causal diagrams must state a directional
relationship because we are theorizing that one variable causes
another.</p>
<p>This is the simplest form of a causal diagram. If we were to
statistically model the effect for this relationship using regression, a
simple two-variable model containing the exposure and the outcome would
give us a correct, unbiased estimate of the actual strength of the cause
on the effect. However, we are rarely that lucky.</p>
<p><strong>Question</strong>: Can you think of any situations that may
bias the observed relationship between the ambulance and the dog
barking?</p>
</div>
<div class="section level2">
<h2 id="making-dags">Making DAGs<a class="anchor" aria-label="anchor" href="#making-dags"></a>
</h2>
<p>There are several options for making DAGs. We could draw them by hand
on a piece of paper or in software that allows us to make diagrams. But
there are specialized options available online and in R we can use
instead.</p>
<div class="section level3">
<h3 id="dagitty">Dagitty<a class="anchor" aria-label="anchor" href="#dagitty"></a>
</h3>
<p>Dagitty offers both an online version (<a href="http://www.dagitty.net/" class="external-link uri">http://www.dagitty.net/</a>)
and an R package. The online version provides a user interface. You can
also export the diagram from the online interface to R.</p>
<p><strong>Demo:</strong> Dagitty online and export to R</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dag_export</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/dagitty/man/dagitty.html" class="external-link">dagitty</a></span><span class="op">(</span> <span class="st">'dag {
bb="0,0,1,1"
age [pos="0.258,0.073"]
genetics [pos="0.416,0.539"]
lung_cancer [outcome,pos="0.607,0.288"]
pollution [pos="0.261,0.178"]
smoking [exposure,pos="0.265,0.291"]
age -&gt; lung_cancer
genetics -&gt; lung_cancer
genetics -&gt; smoking
pollution -&gt; lung_cancer
smoking -&gt; lung_cancer
}'</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">dag_export</span>, height<span class="op">=</span><span class="fl">1</span><span class="op">)</span></code></pre></div>
<p><img src="EpiForBioWorkshop_files/figure-html/unnamed-chunk-3-1.png" width="288"></p>
</div>
<div class="section level3">
<h3 id="ggdag">ggDag<a class="anchor" aria-label="anchor" href="#ggdag"></a>
</h3>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/ggdag/man/ggdag.html" class="external-link">ggdag</a></span><span class="op">(</span><span class="va">dag_export</span>, node_size <span class="op">=</span> <span class="fl">18</span>, text_col<span class="op">=</span><span class="st">"red"</span><span class="op">)</span><span class="op">+</span><span class="fu"><a href="https://rdrr.io/pkg/ggdag/man/theme_dag_blank.html" class="external-link">theme_dag_blank</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="EpiForBioWorkshop_files/figure-html/unnamed-chunk-4-1.png" width="576"></p>
<p><strong>Exercise 1:</strong> Make the DAG for a causal question of
interest. Then export it to R. Using dagitty and ggdag, try adjusting
different aspects of it (position, color).</p>
</div>
</div>
<div class="section level2">
<h2 id="bias-and-confounding">Bias and Confounding<a class="anchor" aria-label="anchor" href="#bias-and-confounding"></a>
</h2>
<div class="section level3">
<h3 id="confounding">Confounding<a class="anchor" aria-label="anchor" href="#confounding"></a>
</h3>
<p>Confounding arises when a third variable is present in the
relationship between cause-and-effect. A confounder is present when:</p>
<ol style="list-style-type: decimal">
<li>The confounder causes the outcome</li>
<li>The confounder causes the exposure</li>
<li>The confounder is not a mediator (i.e. it is not present on the
causal pathway between the exposure and the outcome)</li>
</ol>
<p>Returning to our example, the relationship between the ambulance and
the barking dog could be confounded by a third variable: a nearby car
crash. Perhaps this car crash is what started the dog barking before the
ambulance even arrived, but the ambulance caused the dog to bark even
more.</p>
<p><img src="EpiForBioWorkshop_files/figure-html/unnamed-chunk-5-1.png" width="288"></p>
<p>The pathway between the exposure and the outcome through the
confounder is called a “backdoor pathway” and it can bias the
relationship between the exposure and the outcome you observe. In this
situation, we might observe barking and conclude that it occurred due to
the ambulance when in fact it was due to the car crash. Or we might
conclude that the volume of the barking caused by the ambulance alone is
much greater than it actually is due to the combined effect of both the
car crash and the ambulance.</p>
<p><strong>Question:</strong> What are some common confounders in
bioinformatics or health research?</p>
<p>We can “deconfound” the effect of interest by controlling for the
confounder. The old-fashioned way of doing this is to manually stratify
your data by levels of the confounder. Then you calculate an effect size
for each stratum then calculate a weighted average across strata. You
may have been exposed to this statistical method in a biostatistics
course as a Cochran-Mantel-Haenszel statistic.</p>
<p>In a regression model, we can include the confounder as an
independent variable in our model and it is controlled for. When we
include the confounder as a variable in a regression model, we are
effectively stratifying the effect across different levels of the
confounder then taking the weighted average across strata to get an
effect size.</p>
<p><strong>Exercise 2:</strong> Using your DAG for a causal relationship
of interest to you, consider potential confounders.</p>
<div class="section level4">
<h4 id="unmeasured-confounding">Unmeasured confounding<a class="anchor" aria-label="anchor" href="#unmeasured-confounding"></a>
</h4>
<p>A common issue in observational studies is that there may be
confounders that you are not aware of or did not assess. Unmeasured
confounding is a problem with no easy solution as it is effectively a
missing data problem–we cannot deconfound for a confounder we have not
measured.</p>
<p>While not a full solution, one possible method for considering
unmeasured confounding is to conduct sensitivity analysis to see how
large in magnitude the unmeasured confounder would need to be to
substantially affect the observed effect size. If the effect size is
sensitive to this hypothetical confounder, we should be more careful in
how we interpret our results.</p>
</div>
</div>
<div class="section level3">
<h3 id="colliders">Colliders<a class="anchor" aria-label="anchor" href="#colliders"></a>
</h3>
<p>Taken at face value, colliders are similar to confounders. However,
colliders are not common causes of the exposure and outcome.
Instead:</p>
<ol style="list-style-type: decimal">
<li>The outcome causes the collider</li>
<li>The exposure causes the collider</li>
<li>The collider is not a mediator</li>
</ol>
<p><img src="EpiForBioWorkshop_files/figure-html/unnamed-chunk-6-1.png" width="288"></p>
<p><img src="EpiForBioWorkshop_files/figure-html/unnamed-chunk-7-1.png" width="768"></p>
<p>Adjusting for a collider as one would for a confounder can create
bias. In effect by adjusting for a collider, a backdoor pathway is
opened between the exposure and the outcome through the collider. So
what’s the proper way of dealing with colliders? Ignoring them!
Hypothesized colliders should not be adjusted for or included in
models.</p>
<div class="section level4">
<h4 id="side-note-on-pathways">Side note on pathways<a class="anchor" aria-label="anchor" href="#side-note-on-pathways"></a>
</h4>
<p>In a DAG, a pathway is considered blocked: - If there are two arrows
pointing into each other along the path (as in the case of a collider) -
If you have deconfounded (controlled, stratified, adjusted) for a
variable on the pathway</p>
<p><img src="EpiForBioWorkshop_files/figure-html/unnamed-chunk-8-1.png" width="576"></p>
<p><strong>Exercise 3:</strong> Using the DAG you made in Exercises 1
and 2, add a collider.</p>
</div>
</div>
<div class="section level3">
<h3 id="selection-bias">Selection bias<a class="anchor" aria-label="anchor" href="#selection-bias"></a>
</h3>
<p>Another major issue of causal inference is selection bias. This
arises when selection is dependent on the exposure and the outcome in
the study:</p>
<p><img src="EpiForBioWorkshop_files/figure-html/unnamed-chunk-9-1.png" width="288"></p>
<p>As can be seen from the DAG above, selection bias is a collider. In
this case, the study is conditioned on the collider by the act of
selection for the study. As a result, the study is biased. One
particularly memorable way to think of selection bias is as the “already
dead” problem. People who have already died of the outcome are not alive
to be in the study. These people may have a more aggressive or serious
form of the outcome and not including them masks some of the causal
relationship.</p>
<p>A common source of selection bias is loss to follow-up in many
studies. In survival analysis in particular this can be important as the
people who are lost may differ in important ways from those who
completed the study. Unfortunately because these people were lost, it is
often difficult to assess how they differed from those who were not
lost.</p>
<p><strong>Question:</strong> Is the collider you added in Exercise 3 a
possible source of selection bias? If so, how?</p>
</div>
</div>
<div class="section level2">
<h2 id="study-designs">Study Designs<a class="anchor" aria-label="anchor" href="#study-designs"></a>
</h2>
<div class="section level3">
<h3 id="randomized-control-trials-rcts">Randomized Control Trials (RCTs)<a class="anchor" aria-label="anchor" href="#randomized-control-trials-rcts"></a>
</h3>
<ul>
<li>Exposure of interest is assigned at random</li>
<li>Because exposure is assigned at random, there are no possible
confounders of the relationship between exposure and outcome</li>
<li>Considered the gold standard for causal inference</li>
<li>Selection bias is still possible</li>
</ul>
<p><strong>Question:</strong> How can selection bias occur in an
RCT?</p>
<div class="section level4">
<h4 id="instrumental-variable-analysismendelian-randomization">Instrumental Variable Analysis/Mendelian Randomization<a class="anchor" aria-label="anchor" href="#instrumental-variable-analysismendelian-randomization"></a>
</h4>
<ul>
<li>Attempts to mimic an RCT using a source of pseudorandomization</li>
</ul>
<p>Consider the DAG for an RCT:</p>
<p><img src="EpiForBioWorkshop_files/figure-html/unnamed-chunk-10-1.png" width="480"></p>
<p>In IV/MR analysis the coin flip is replaced with a source of
randomization such as genetic variation:</p>
<p><img src="EpiForBioWorkshop_files/figure-html/unnamed-chunk-11-1.png" width="480"></p>
<p>Researchers have used a variety of factors as sources of
pseudorandomness in IV studies:</p>
<ul>
<li><p>Before and after implementation of a new government
policy</p></li>
<li><p>Distance from a medical center</p></li>
<li><p>Genetic variants</p></li>
</ul>
<p><strong>Question:</strong> IV/MR studies can seem quite powerful
because they allow us to make stronger causal arguments without doing an
RCT. What might be some issues with doing IV/MR studies?</p>
</div>
</div>
<div class="section level3">
<h3 id="cohort-studies">Cohort Studies<a class="anchor" aria-label="anchor" href="#cohort-studies"></a>
</h3>
<ul>
<li>Participants are chosen on a common characteristic (such as
occupation, location, or history of a particular disease)</li>
<li>They are then followed over time and exposures and outcomes can be
assessed</li>
<li>Selection bias and confounding can occur</li>
</ul>
</div>
<div class="section level3">
<h3 id="case-control-studies">Case-Control Studies<a class="anchor" aria-label="anchor" href="#case-control-studies"></a>
</h3>
<ul>
<li>Participants are chosen based on having an outcome of interest
(cases)</li>
<li>Then controls are selected from a similar population to the cases
(often matched on age, sex, and other demographic variables)</li>
<li>Selection bias and confounding can occur</li>
</ul>
</div>
<div class="section level3">
<h3 id="genome-wide-association-study-gwas">Genome-Wide Association Study (GWAS)<a class="anchor" aria-label="anchor" href="#genome-wide-association-study-gwas"></a>
</h3>
<ul>
<li><p>Observational study of phenotypes</p></li>
<li><p>Often participants are chosen based on a specific phenotype or
disease</p></li>
<li><p>Can be a case-control or a cohort study</p></li>
</ul>
<p><strong>Question</strong>: What are some possible sources of
confounding or bias in a GWAS?</p>
</div>
</div>
<div class="section level2">
<h2 id="causal-inference-in-r">Causal Inference in R<a class="anchor" aria-label="anchor" href="#causal-inference-in-r"></a>
</h2>
<p>R provides many packages that are helpful in causal inference.</p>
<div class="section level3">
<h3 id="dagitty-1">Dagitty<a class="anchor" aria-label="anchor" href="#dagitty-1"></a>
</h3>
<p>Dagitty allows for creating causal diagrams, but also gives you more
information including what variables to adjust for in complicated causal
models.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">g1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/dagitty/man/dagitty.html" class="external-link">dagitty</a></span><span class="op">(</span> <span class="st">"dag {
    W1 -&gt; Z1 -&gt; X -&gt; Y
    Z1 &lt;- V -&gt; Z2
    W2 -&gt; Z2 -&gt; Y
    X &lt;- W1 -&gt; W2 -&gt; Y
}"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/dagitty/man/graphLayout.html" class="external-link">graphLayout</a></span><span class="op">(</span><span class="va">g1</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="EpiForBioWorkshop_files/figure-html/unnamed-chunk-12-1.png" width="576"></p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">for</span><span class="op">(</span> <span class="va">n</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">g1</span><span class="op">)</span> <span class="op">)</span><span class="op">{</span>
    <span class="kw">for</span><span class="op">(</span> <span class="va">m</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/setops-methods.html" class="external-link">setdiff</a></span><span class="op">(</span> <span class="fu"><a href="https://rdrr.io/pkg/dagitty/man/AncestralRelations.html" class="external-link">descendants</a></span><span class="op">(</span> <span class="va">g1</span>, <span class="va">n</span> <span class="op">)</span>, <span class="va">n</span> <span class="op">)</span> <span class="op">)</span><span class="op">{</span>
        <span class="va">a</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/dagitty/man/adjustmentSets.html" class="external-link">adjustmentSets</a></span><span class="op">(</span> <span class="va">g1</span>, <span class="va">n</span>, <span class="va">m</span> <span class="op">)</span>
        <span class="kw">if</span><span class="op">(</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">a</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0</span> <span class="op">)</span><span class="op">{</span>
            <span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"The total effect of "</span>,<span class="va">n</span>,<span class="st">" on "</span>,<span class="va">m</span>,
                <span class="st">" is identifiable controlling for:\n"</span>,sep<span class="op">=</span><span class="st">""</span><span class="op">)</span>
            <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span> <span class="va">a</span>, prefix<span class="op">=</span><span class="st">" * "</span> <span class="op">)</span>
        <span class="op">}</span>
    <span class="op">}</span>
<span class="op">}</span></code></pre></div>
<pre><code><span class="co">## The total effect of V on Z2 is identifiable controlling for:</span>
<span class="co">##  *  {}</span>
<span class="co">## The total effect of V on Y is identifiable controlling for:</span>
<span class="co">##  *  {}</span>
<span class="co">## The total effect of V on Z1 is identifiable controlling for:</span>
<span class="co">##  *  {}</span>
<span class="co">## The total effect of V on X is identifiable controlling for:</span>
<span class="co">##  *  {}</span>
<span class="co">## The total effect of W1 on Z1 is identifiable controlling for:</span>
<span class="co">##  *  {}</span>
<span class="co">## The total effect of W1 on X is identifiable controlling for:</span>
<span class="co">##  *  {}</span>
<span class="co">## The total effect of W1 on Y is identifiable controlling for:</span>
<span class="co">##  *  {}</span>
<span class="co">## The total effect of W1 on W2 is identifiable controlling for:</span>
<span class="co">##  *  {}</span>
<span class="co">## The total effect of W1 on Z2 is identifiable controlling for:</span>
<span class="co">##  *  {}</span>
<span class="co">## The total effect of W2 on Z2 is identifiable controlling for:</span>
<span class="co">##  *  {}</span>
<span class="co">## The total effect of W2 on Y is identifiable controlling for:</span>
<span class="co">## { V, X }</span>
<span class="co">## { W1 }</span>
<span class="co">## The total effect of X on Y is identifiable controlling for:</span>
<span class="co">## { W2, Z2 }</span>
<span class="co">## { V, W2 }</span>
<span class="co">## { V, W1 }</span>
<span class="co">## { W1, Z1 }</span>
<span class="co">## The total effect of Z1 on X is identifiable controlling for:</span>
<span class="co">## { W1 }</span>
<span class="co">## The total effect of Z1 on Y is identifiable controlling for:</span>
<span class="co">## { W1, W2, Z2 }</span>
<span class="co">## { V, W1 }</span>
<span class="co">## The total effect of Z2 on Y is identifiable controlling for:</span>
<span class="co">## { W2, X }</span>
<span class="co">## { W1, W2, Z1 }</span>
<span class="co">## { V, W2 }</span></code></pre>
</div>
<div class="section level3">
<h3 id="ggdag-1">ggdag<a class="anchor" aria-label="anchor" href="#ggdag-1"></a>
</h3>
<p>Another option for making causal diagrams is ggdag which can accept
daggity objects and plots DAGs using ggplot2.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/ggdag/man/adjustment_sets.html" class="external-link">ggdag_adjustment_set</a></span><span class="op">(</span><span class="va">g1</span>, <span class="st">"X"</span>, <span class="st">"Y"</span><span class="op">)</span></code></pre></div>
<p><img src="EpiForBioWorkshop_files/figure-html/unnamed-chunk-14-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="adjusting-for-confounders">Adjusting for confounders<a class="anchor" aria-label="anchor" href="#adjusting-for-confounders"></a>
</h3>
<div class="section level4">
<h4 id="deseq2">DESeq2<a class="anchor" aria-label="anchor" href="#deseq2"></a>
</h4>
<p>DESeq2 uses negative binomial regression. As such, it allows you to
deconfound by including the confounder as a variable in the design
statement:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ddsSE</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeqDataSet.html" class="external-link">DESeqDataSet</a></span><span class="op">(</span><span class="va">se</span>, design <span class="op">=</span> <span class="op">~</span> <span class="va">exposure</span> <span class="op">+</span> <span class="va">confounder1</span> <span class="op">+</span> <span class="va">confounder2</span><span class="op">)</span>
<span class="va">ddsSE</span></code></pre></div>
</div>
<div class="section level4">
<h4 id="edger-and-limma">edgeR and limma<a class="anchor" aria-label="anchor" href="#edger-and-limma"></a>
</h4>
<p>You can use <code><a href="https://rdrr.io/r/stats/model.matrix.html" class="external-link">model.matrix()</a></code> to specify the exposure and
confounders:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">batch</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">ds</span><span class="op">$</span><span class="va">Batch</span><span class="op">)</span>
<span class="va">treat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">ds</span><span class="op">$</span><span class="va">tx</span><span class="op">)</span>
<span class="va">y</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">ds</span><span class="op">$</span><span class="va">surv</span><span class="op">)</span>
<span class="va">design</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html" class="external-link">model.matrix</a></span><span class="op">(</span><span class="op">~</span><span class="va">treat</span> <span class="op">+</span> <span class="va">batch</span><span class="op">)</span></code></pre></div>
<p>Then fit the model either in edgeR:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">fit</span> <span class="op">&lt;-</span> <span class="fu">glmQLFit</span><span class="op">(</span><span class="va">y</span>, <span class="va">design</span><span class="op">)</span></code></pre></div>
<p>Or in limma:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">fit</span> <span class="op">&lt;-</span> <span class="fu">lmFit</span><span class="op">(</span><span class="va">y</span>, <span class="va">design</span><span class="op">)</span></code></pre></div>
<p>As mentioned in the confounding section, estimates obtained for the
relationship between the exposure and the outcome have been adjusted for
the included confounders (deconfounded). This means that the observed
estimate is an averaged effect size across strata of the
confounder(s).</p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="exercise-in-r-using-curatedmetagenomicdata-and-deseq2">Exercise in R using curatedMetagenomicData and DESeq2<a class="anchor" aria-label="anchor" href="#exercise-in-r-using-curatedmetagenomicdata-and-deseq2"></a>
</h2>
<p>We will be using the VogtmannE 2016 dataset in
curatedMetagenomicData. The dataset has 52 cases of colorectal cancer
(crc) and 52 controls. First, let’s create a causal diagram for this
relationship.</p>
<ol style="list-style-type: decimal">
<li><p>We are interested on whether our exposure (colorectal cancer)
changes the outcome (gut microbiome). Create a DAG for this
relationship. Consider possible confounders, selection bias, etc. Using
the DAG, determine what to adjust for.</p></li>
<li><p>In R, download the data and look at study condition. Remove any
missing values.</p></li>
<li><p>Run a bivariate model of the exposure and the outcome without any
deconfounding. What bacterial taxa are significant?</p></li>
<li><p>Deconfound for confounders you identified in your DAG. If you
can’t deconfound for them because they were not measured, how does that
affect your results?</p></li>
</ol>
<p>Install curatedMetagenomicData</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>
    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span>

<span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"curatedMetagenomicData"</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/waldronlab/curatedMetagenomicData" class="external-link">curatedMetagenomicData</a></span><span class="op">)</span></code></pre></div>
<p>Draw our assumptions. For this example, we use the VogtmannE 2016
dataset in cMD. The dataset has 52 cases of colorectal cancer (crc) and
52 controls. First, let’s create a causal diagram for this
relationship.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">v_dag</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/dagitty/man/dagitty.html" class="external-link">dagitty</a></span><span class="op">(</span> <span class="st">"dag {
  crc -&gt; microbiome
  crc &lt;- age -&gt; microbiome
  crc &lt;- gender -&gt; microbiome
}"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/ggdag/man/ggdag.html" class="external-link">ggdag</a></span><span class="op">(</span><span class="va">v_dag</span>, text_col<span class="op">=</span><span class="st">"red"</span><span class="op">)</span><span class="op">+</span><span class="fu"><a href="https://rdrr.io/pkg/ggdag/man/theme_dag_blank.html" class="external-link">theme_dag</a></span><span class="op">(</span><span class="op">)</span><span class="op">+</span><span class="fu"><a href="https://rdrr.io/pkg/ggdag/man/theme_dag_grey.html" class="external-link">theme_dag_gray</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="EpiForBioWorkshop_files/figure-html/unnamed-chunk-20-1.png" width="700"></p>
<p>What do we need to adjust for?</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/ggdag/man/adjustment_sets.html" class="external-link">ggdag_adjustment_set</a></span><span class="op">(</span><span class="va">v_dag</span>, exposure <span class="op">=</span> <span class="st">"crc"</span>, outcome <span class="op">=</span> <span class="st">"microbiome"</span>, text_col<span class="op">=</span><span class="st">"purple"</span><span class="op">)</span><span class="op">+</span><span class="fu"><a href="https://rdrr.io/pkg/ggdag/man/theme_dag_grey.html" class="external-link">theme_dag_gray</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="EpiForBioWorkshop_files/figure-html/unnamed-chunk-21-1.png" width="700"></p>
<p>Download the data, extract the expressionset, and only take responses
for which we have a value for study condition.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">vogtmann</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/curatedMetagenomicData/man/curatedMetagenomicData.html" class="external-link">curatedMetagenomicData</a></span><span class="op">(</span><span class="st">"2021-03-31.VogtmannE_2016.relative_abundance"</span>, dryrun <span class="op">=</span> <span class="cn">FALSE</span>, counts <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>

<span class="va">vm</span> <span class="op">&lt;-</span> <span class="va">vogtmann</span><span class="op">$</span><span class="va">`2021-03-31.VogtmannE_2016.relative_abundance`</span>

<span class="va">vm.sub</span> <span class="op">&lt;-</span> <span class="va">vm</span><span class="op">[</span>,<span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">vm</span><span class="op">$</span><span class="va">study_condition</span><span class="op">)</span><span class="op">]</span></code></pre></div>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">vm.sub</span><span class="op">$</span><span class="va">study_condition</span>, useNA<span class="op">=</span><span class="st">"ifany"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## </span>
<span class="co">## control     CRC </span>
<span class="co">##      52      52</span></code></pre>
<p>First we conduct a basic analysis to examine whether there are any
statistically significant differences in taxa of microbiota by study
condition. We find 172 significant taxa in this crude analysis.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#vm.pseq &lt;- ExpressionSet2phyloseq(vm.sub)</span>
<span class="va">phylo</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mia/man/makePhyloseqFromTreeSummarizedExperiment.html" class="external-link">makePhyloseqFromTreeSummarizedExperiment</a></span><span class="op">(</span><span class="va">vm.sub</span>, abund_values <span class="op">=</span> <span class="st">"relative_abundance"</span><span class="op">)</span>
<span class="va">deseqds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/phyloseq_to_deseq2.html" class="external-link">phyloseq_to_deseq2</a></span><span class="op">(</span><span class="va">phylo</span>, <span class="op">~</span><span class="va">study_condition</span><span class="op">)</span>
<span class="va">dx_only</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeq.html" class="external-link">DESeq</a></span><span class="op">(</span><span class="va">deseqds</span>, test<span class="op">=</span><span class="st">"Wald"</span>, fitType<span class="op">=</span><span class="st">"parametric"</span><span class="op">)</span>
<span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/results.html" class="external-link">results</a></span><span class="op">(</span><span class="va">dx_only</span>, cooksCutoff <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="va">alpha</span> <span class="op">&lt;-</span> <span class="fl">0.05</span>
<span class="va">sigtab</span> <span class="op">&lt;-</span> <span class="va">res</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">res</span><span class="op">$</span><span class="va">padj</span> <span class="op">&lt;</span> <span class="va">alpha</span><span class="op">)</span>, <span class="op">]</span>
<span class="va">sigtab</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="fu">as</span><span class="op">(</span><span class="va">sigtab</span>, <span class="st">"data.frame"</span><span class="op">)</span>, <span class="fu">as</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html" class="external-link">tax_table</a></span><span class="op">(</span><span class="va">phylo</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">sigtab</span><span class="op">)</span>, <span class="op">]</span>, <span class="st">"matrix"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">sigtab</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] 172</span></code></pre>
<p>Next we adjust (deconfound) for categorized age. We find 175
significant results:</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dx_age</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/phyloseq_to_deseq2.html" class="external-link">phyloseq_to_deseq2</a></span><span class="op">(</span><span class="va">phylo</span>, <span class="op">~</span> <span class="va">study_condition</span> <span class="op">+</span> <span class="va">age_category</span><span class="op">)</span>
<span class="va">dx_age</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeq.html" class="external-link">DESeq</a></span><span class="op">(</span><span class="va">dx_age</span>, test<span class="op">=</span><span class="st">"Wald"</span>, fitType<span class="op">=</span><span class="st">"parametric"</span><span class="op">)</span>
<span class="va">res</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/results.html" class="external-link">results</a></span><span class="op">(</span><span class="va">dx_age</span>, cooksCutoff <span class="op">=</span> <span class="cn">FALSE</span>, contrast<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"study_condition"</span>, <span class="st">"CRC"</span>, <span class="st">"control"</span><span class="op">)</span><span class="op">)</span>
<span class="va">alpha</span> <span class="op">=</span> <span class="fl">0.05</span>
<span class="va">sigtab</span> <span class="op">=</span> <span class="va">res</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">res</span><span class="op">$</span><span class="va">padj</span> <span class="op">&lt;</span> <span class="va">alpha</span><span class="op">)</span>, <span class="op">]</span>
<span class="va">sigtab</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="fu">as</span><span class="op">(</span><span class="va">sigtab</span>, <span class="st">"data.frame"</span><span class="op">)</span>, <span class="fu">as</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html" class="external-link">tax_table</a></span><span class="op">(</span><span class="va">phylo</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">sigtab</span><span class="op">)</span>, <span class="op">]</span>, <span class="st">"matrix"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">sigtab</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] 175</span></code></pre>
<p>Finally we add in gender. Again observing 183 significant
results.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dx_age_gender</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/phyloseq_to_deseq2.html" class="external-link">phyloseq_to_deseq2</a></span><span class="op">(</span><span class="va">phylo</span>, <span class="op">~</span> <span class="va">study_condition</span> <span class="op">+</span> <span class="va">age_category</span> <span class="op">+</span> <span class="va">gender</span><span class="op">)</span>
<span class="va">dx_age_gender</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeq.html" class="external-link">DESeq</a></span><span class="op">(</span><span class="va">dx_age_gender</span>, test<span class="op">=</span><span class="st">"Wald"</span>, fitType<span class="op">=</span><span class="st">"parametric"</span><span class="op">)</span>
<span class="va">res</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/results.html" class="external-link">results</a></span><span class="op">(</span><span class="va">dx_age_gender</span>, cooksCutoff <span class="op">=</span> <span class="cn">FALSE</span>, contrast<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"study_condition"</span>, <span class="st">"CRC"</span>, <span class="st">"control"</span><span class="op">)</span><span class="op">)</span>
<span class="va">alpha</span> <span class="op">=</span> <span class="fl">0.05</span>
<span class="va">sigtab</span> <span class="op">=</span> <span class="va">res</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">res</span><span class="op">$</span><span class="va">padj</span> <span class="op">&lt;</span> <span class="va">alpha</span><span class="op">)</span>, <span class="op">]</span>
<span class="va">sigtab</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="fu">as</span><span class="op">(</span><span class="va">sigtab</span>, <span class="st">"data.frame"</span><span class="op">)</span>, <span class="fu">as</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html" class="external-link">tax_table</a></span><span class="op">(</span><span class="va">phylo</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">sigtab</span><span class="op">)</span>, <span class="op">]</span>, <span class="st">"matrix"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">sigtab</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] 183</span></code></pre>
<p>As mentioned in the confounding section, estimates obtained for the
relationship between the exposure and the outcome have been adjusted for
the included confounders (deconfounded). This means that the observed
estimate is an averaged effect size across strata of the
confounder(s).</p>
<p><strong>Question:</strong> After adjusting for confounding by age and
gender, we found that colorectal cancer is associated with increased
abundance of two taxa, but we would probably still hesitate to argue
that we’ve shown that colorectal cancer causes these changes in
abundance. What sources of bias might still potentially affect our
model? How could we design a better study to account for these
biases?</p>
</div>
<div class="section level2">
<h2 id="discussion-questions">Discussion Questions<a class="anchor" aria-label="anchor" href="#discussion-questions"></a>
</h2>
<ol style="list-style-type: decimal">
<li>Often in our work, we are presented with data that have already been
collected and we are asked to analyze it. Other times we have the
privilege of designing a study from scratch. And some times we are
brought in when the analyses are nearly done for a paper and just asked
for our feedback. When during the study process should we create a DAG?
Is it too late to make a DAG if the analyses are nearly completed?</li>
<li>One important subject that I did not touch on in this workshop was
generalizability or transportability of causal findings (applying the
results to an external population). What issues might arise when doing
so? How can a DAG be used to resolve issues of generalizability and
transportability?</li>
</ol>
</div>
<div class="section level2">
<h2 id="conclusions">Conclusions<a class="anchor" aria-label="anchor" href="#conclusions"></a>
</h2>
<p>Unfortunately there’s no silver bullet for confounding or bias in R
or Bioconductor. Instead, we as researchers must carefully consider
possible sources of bias–ideally as we begin designing the study. DAGs
are helpful in providing a clear visualization of the hypothesized
causal mechanism and identifying potential confounders. As Hernan says,
“Draw your assumptions before you draw your conclusions.”</p>
<p>Unmeasured confounding presents a particularly difficult problem for
causal inference. Because the confounder was not measured, we cannot
easily deconfound for it.</p>
<p>It is important for us to remember that the definition of a cause is
rooted in the counterfactual: we need to find a way to come as close as
possible to observing what would have happened if a participant’s
exposure status had been different.</p>
</div>
<div class="section level2">
<h2 id="further-readingmaterials">Further Reading/Materials<a class="anchor" aria-label="anchor" href="#further-readingmaterials"></a>
</h2>
<p>This has been a basic primer on causal inference but if you’d like to
learn more:</p>
<ul>
<li>Pearl and Mackenzie <em>Book of Why: The New Science of Causal
Effect</em> (easy, fun read!)</li>
<li>Hernan and Robins <em>Causal Inference</em> available free online:
<a href="https://www.hsph.harvard.edu/miguel-hernan/causal-inference-book/" class="external-link uri">https://www.hsph.harvard.edu/miguel-hernan/causal-inference-book/</a>
</li>
<li>Pearl <em>Causality: Models, Reasoning, and Inference</em>
</li>
<li>Hernan’s edX course available free online: <a href="https://www.edx.org/course/causal-diagrams-draw-your-assumptions-before-your" class="external-link uri">https://www.edx.org/course/causal-diagrams-draw-your-assumptions-before-your</a>
</li>
</ul>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Levi Waldron.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
